runtime/vm/vmprops.c

```diff
@@ -530,6 +530,103 @@ getLibSubDir(J9JavaVM *vm, const char *subDir, char **value)
 }
 
 #define COM_SUN_MANAGEMENT "com.sun.management"
+
+UDATA
+loadCMDLineSystemProperties(J9JavaVM *vm, JavaVMInitArgs *initArgs)
+{
+	UDATA rc = J9SYSPROP_ERROR_NONE;
+	BOOLEAN addManagementModule = FALSE;
+	jint i = 0;
+	const char* propValue = NULL;
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
+	/* Create the -D properties. This may override any of the writeable properties above.
+	 * Should the command line override read-only props?
+	 */
+	for (i = 0; i < initArgs->nOptions; ++i) {
+		char * optionString = initArgs->options[i].optionString;
+
+		if (0 == strncmp("-D", optionString, 2)) {
+			J9VMSystemProperty *currentProp = NULL;
+			char *propNameCopy = NULL;
+			char *propValueCopy = NULL;
+			UDATA propNameLen = 0;
+
+			propValue = strchr(optionString + 2, '=');
+			if (NULL == propValue) {
+				propNameLen = strlen(optionString) - 2;
+				propValue = optionString + 2 + propNameLen;
+			} else {
+				propNameLen = propValue - (optionString + 2);
+				++propValue;
+			}
+
+			{
+				UDATA valueLength = strlen(propValue);
+
+				propNameCopy = (char *)getMUtf8String(vm, optionString + 2, propNameLen); /* get a copy of the property name */
+				if (NULL == propNameCopy) {
+					rc = J9SYSPROP_ERROR_OUT_OF_MEMORY;
+					goto fail;
+				}
+				propValueCopy = (char *)getMUtf8String(vm, propValue, valueLength); /* get a copy of the property value */
+				if (NULL == propValueCopy) {
+					j9mem_free_memory(propNameCopy);
+					rc = J9SYSPROP_ERROR_OUT_OF_MEMORY;
+					goto fail;
+				}
+			}
+#if JAVA_SPEC_VERSION >= 11
+				/* defining any system property starting with "com.sun.management" should
+				 * cause the jdk.management.agent module to be loaded. This occurs in
+				 * jclcinit.c:initializeRequiredClasses().
+				 */
+				if (!addManagementModule && (0 == strncmp(propNameCopy, COM_SUN_MANAGEMENT, strlen(COM_SUN_MANAGEMENT)))) {
+					vm->extendedRuntimeFlags2 |= J9_EXTENDED_RUNTIME2_LOAD_AGENT_MODULE;
+					addManagementModule = TRUE;
+				} else if ('\0' != propValue[0]) {
+					/* Support for java.endorsed.dirs and java.ext.dirs is disabled in Java 9+. */
+					if (0 == strncmp(JAVA_ENDORSED_DIRS, propNameCopy, sizeof(JAVA_ENDORSED_DIRS))) {
+						j9nls_printf(PORTLIB, J9NLS_ERROR, J9NLS_VM_PROPERTY_JAVA_ENDORSED_DIR_UNSUPPORTED, optionString + 2);
+						j9mem_free_memory(propNameCopy);
+						j9mem_free_memory(propValueCopy);
+						rc = J9SYSPROP_ERROR_UNSUPPORTED_PROP;
+						goto fail;
+					}
+					if (0 == strncmp(JAVA_EXT_DIRS, propNameCopy, sizeof(JAVA_EXT_DIRS))) {
+						j9nls_printf(PORTLIB, J9NLS_ERROR, J9NLS_VM_PROPERTY_JAVA_EXT_DIR_UNSUPPORTED, optionString + 2);
+						j9mem_free_memory(propNameCopy);
+						j9mem_free_memory(propValueCopy);
```

Comment:
These `j9mem_free_memory()` calls can be consolated before `return rc;` if `propNameCopy` or `propValueCopy` are not `NULL`. It seems the existing code missed cleanup in some cases.

Reply:
https://github.com/eclipse-openj9/openj9/commit/22bc21d406bacd4ec2a26c64f3a76bab7b5d33fc did consolidation of `j9mem_free_memory()`, also the changes moving the comments, any chance it was overwritten accidentally?